state/etcdstatedriver.go

```diff
@@ -211,7 +215,11 @@ func (d *EtcdStateDriver) ReadAllState(baseKey string, sType core.State,
 	return ReadAllStateCommon(d, baseKey, sType, unmarshal)
 }
 
-func (d *EtcdStateDriver) channelStateEvents(sType core.State,
+// ChannelStateEvents watches for updates(created, modify, delete) to a state of
+// specified type and unmarshals (given a function) all changes and puts then on
+// channel of core.WatchState objects.
+// XXX: move this to some common file
+func ChannelStateEvents(d core.StateDriver, sType core.State,
```

Comment:
Is this a new public function exposed for the state driver now, which was embedded into WatchAll earlier, but I saw your comment about why Watchall can miss some events due to timing. Please clarify.